stores/sql/mysql/chain.go

```diff
@@ -0,0 +1,314 @@
+				time.Now().UTC(),
+				ssql.Hash256(e.ID),
+				e.StateElement.LeafIndex,
+				ssql.MerkleProof{Hashes: e.StateElement.MerkleProof},
+				ssql.Currency(e.SiacoinOutput.Value),
+				ssql.Hash256(e.SiacoinOutput.Address),
+				e.MaturityHeight,
+			); err != nil {
+				return fmt.Errorf("failed to insert unspent output: %w", err)
+			}
+		}
+	}
+
+	// remove events created at the reverted index
+	res, err := c.tx.Exec(c.ctx, "DELETE FROM wallet_events WHERE height = ? AND block_id = ?", index.Height, ssql.Hash256(index.ID))
+	if err != nil {
+		return fmt.Errorf("failed to delete events: %w", err)
+	} else if n, err := res.RowsAffected(); err != nil {
+		return fmt.Errorf("failed to get rows affected: %w", err)
+	} else if n > 0 {
+		c.l.Debugw(fmt.Sprintf("removed %d events", n), "height", index.Height, "block_id", index.ID)
+	}
+	return nil
+}
+
+func (c ChainUpdateTx) UpdateChainIndex(index types.ChainIndex) error {
+	return ssql.UpdateChainIndex(c.ctx, c.tx, index, c.l)
+}
+
+func (c ChainUpdateTx) UpdateContract(fcid types.FileContractID, revisionHeight, revisionNumber, size uint64) error {
+	return ssql.UpdateContract(c.ctx, c.tx, fcid, revisionHeight, revisionNumber, size, c.l)
+}
+
+func (c ChainUpdateTx) UpdateContractProofHeight(fcid types.FileContractID, proofHeight uint64) error {
+	return ssql.UpdateContractProofHeight(c.ctx, c.tx, fcid, proofHeight, c.l)
+}
+
+func (c ChainUpdateTx) UpdateContractState(fcid types.FileContractID, state api.ContractState) error {
+	return ssql.UpdateContractState(c.ctx, c.tx, fcid, state, c.l)
+}
+
+func (c ChainUpdateTx) UpdateFailedContracts(blockHeight uint64) error {
+	return ssql.UpdateFailedContracts(c.ctx, c.tx, blockHeight, c.l)
+}
+
+func (c ChainUpdateTx) UpdateHost(hk types.PublicKey, ha chain.HostAnnouncement, bh uint64, blockID types.BlockID, ts time.Time) error { //
+	c.l.Debugw("update host", "hk", hk, "netaddress", ha.NetAddress)
+
+	// create the announcement
+	if _, err := c.tx.Exec(c.ctx,
+		"INSERT IGNORE INTO host_announcements (created_at, host_key, block_height, block_id, net_address) VALUES (?, ?, ?, ?, ?)",
+		time.Now().UTC(),
+		ssql.PublicKey(hk),
+		bh,
+		blockID.String(),
+		ha.NetAddress,
+	); err != nil {
+		return fmt.Errorf("failed to insert host announcement: %w", err)
+	}
+
+	// create the host
+	var hostID int64
+	if res, err := c.tx.Exec(c.ctx, `
+	INSERT INTO hosts (created_at, public_key, settings, price_table, total_scans, last_scan, last_scan_success, second_to_last_scan_success, scanned, uptime, downtime, recent_downtime, recent_scan_failures, successful_interactions, failed_interactions, lost_sectors, last_announcement, net_address)
+	VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+	ON DUPLICATE KEY UPDATE
+		last_announcement = VALUES(last_announcement),
+		net_address = VALUES(net_address),
+		id = last_insert_id(id)
+	`,
+		time.Now().UTC(),
+		ssql.PublicKey(hk),
+		ssql.Settings{},
+		ssql.PriceTable{},
+		0,
```

Comment:
Can we not rely on the defaults here?

Reply:
No, they're not properly defaulted. It was either a migration or specifying these fields on insert. Since two fields are JSON fields (settings and pts) and prior to MySQL 8.0.13, a JSON column cannot have a non-NULL default value, I opted to avoid the migration. I'm fine changing this though I kind of hate it though, I added these fields one by one running into panics testing this on resyncs.

Reply:
Sad stuff. One day then.